cmd/server/stopwords.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"strings"
+
+	"github.com/moov-io/watchman/pkg/ofac"
+
+	"github.com/abadojack/whatlanggo"
+	"github.com/bbalet/stopwords"
+	"github.com/pariz/gountries"
+)
+
+func removeStopwords(in string, lang whatlanggo.Lang) string {
+	return strings.TrimSpace(stopwords.CleanString(strings.ToLower(in), lang.Iso6391(), false))
+}
+
+// detectLanguage will return a guess as to the appropiate language a given SDN's name
+// is written in. The addresses must be linked to the SDN whose name is detected.
+func detectLanguage(in string, addrs []*ofac.Address) whatlanggo.Lang {
+	info := whatlanggo.Detect(in)
+	if info.IsReliable() {
+		// Return the detected language if whatlanggo is confident enough
+		return info.Lang
+	}
+
+	if len(addrs) == 0 {
+		// If no addresses are associated to this text blob then fallback to English
+		return whatlanggo.Eng
+	}
+
+	// Return the countries primary langauge associated to the primary address for this SDN.
+	//
+	// TODO(adam): Should we do this only if there's one address? If there are multiple should we
+	// fallback to English or a mixed set?
```

Comment:
I don't know that it can be reliably assumed that a given name (`in`) will be in the associated country's official language. Plenty of businesses around the world register with English names rather then their native language.

I think the following approach would be reasonable:
- attempt lang detection
  - if detected language is spoken in any of the countries in `addr`, bump up the confidence level to just above the threshold.
  - if lang threshold is still below threshold, default to a mixed language collection

just my $0.02 :)

Reply:
I like this idea. https://github.com/moov-io/watchman/pull/200/commits/dfbf9c73d73c689ab9099e0f227c1ef9ab7e1829 adopts this logic.